File: urbanfit/lib/models/training.dart
import 'package:urbanfit/models/training_exercise.dart';

class Training {
  final int id;
  final String name;
  final DateTime createdAt;
  final List<TrainingExercise> exercises;
  final double completionPercentage; // Новое поле

  Training({
    required this.id,
    required this.name,
    required this.createdAt,
    required this.exercises,
    required this.completionPercentage, // В конструкторе
  });

  factory Training.fromJson(Map<String, dynamic> json) {
    return Training(
      id: json['id'],
      name: json['name'],
      createdAt: DateTime.parse(json['createdAt']),
      exercises: (json['exercises'] as List?)
              ?.map((e) => TrainingExercise.fromJson(e))
              .toList() ??
          [],
      completionPercentage: (json['completionPercentage'] ?? 0).toDouble(),
    );
  }
}

File: FitAppAPI/FitAppAPI/Models/Training.cs

using System.ComponentModel.DataAnnotations;

namespace FitAppAPI.Models
{
    public class Training
    {
        [Key]
        public int Id { get; set; } 

        [Required]
        public string Name { get; set; }

        [Required]
        public Guid UserId { get; set; } // ID владельца
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public bool IsDraft { get; set; } = true; // Новое поле

        public List<TrainingExercise> TrainingExercises { get; set; } = new();

        public decimal CompletionPercentage { get; set; }

        public ICollection<TrainingSession> Sessions { get; set; }
    }

}
